Hunting Queries/BehaviorAnalytics/Anomalous Account Creation.yaml

```diff
@@ -0,0 +1,26 @@
+id: 0454c8d8-d0a6-42a4-8d03-f5b4fdcbd173
+name: Anomalous Account Creation
+description: |
+  'All users performing user creation for the first time and the target users that were created.'
+requiredDataConnectors:
+  - connectorId: BehaviorAnalytics
+    dataTypes:
+       - BehaviorAnalytics
+       - AuditLogs
+tactics:
+  - Persistence
+relevantTechniques:
+  - T1136
+query: |
+
+ BehaviorAnalytics
+ | where ActionType == "Add user"
+ | where ActivityInsights.FirstTimeUserPerformedAction == True or ActivityInsights.FirstTimeTenantPerformedAction == True
+ | join(
+ AuditLogs
+ ) on $left.SourceRecordId == $right._ItemId
+ | mv-expand TargetResources
+ | extend Target = tostring(TargetResources.userPrincipalName)

```

Comment:
We should fix the parsing of the UserName, UserPrincipalName and TargetUser field for easy connection to additional data types and future use in the Investigation Graph.  UserName comes thru as <username> but in some cases comes thru as <username>_domain.com#EXT#.   You can parse these fields when it has #EXT# in it using this example - | extend TargetUserName = replace("_","@",tostring(split(TargetUserPrincipalName, "#")[0]))

Reply:
Done

Reply:
added to all relevant queries